src/Symfony/Component/Config/Definition/Configuration.php

```diff
@@ -31,8 +31,12 @@ public function __construct(
     ) {
     }
 
+    /**
+     * @return TreeBuilder<'array'>
+     */
     public function getConfigTreeBuilder(): TreeBuilder
     {
+        /** @var TreeBuilder<'array'> $treeBuilder */
         $treeBuilder = new TreeBuilder($this->alias);
```

Comment:
would this make it easier for SA?
```suggestion
        $treeBuilder = new TreeBuilder($this->alias, 'array');
```